Mathlib/LinearAlgebra/AffineSpace/Combination.lean

```diff
@@ -951,3 +951,21 @@ def weightedVSubOfPoint (w : ι → k) : (ι → P) × P →ᵃ[k] V where
      vadd_vsub_assoc, ← sub_add_eq_add_sub, smul_add, Finset.sum_add_distrib]
 
 end AffineMap
+
+namespace AffineEquiv
+
+variable {k : Type*} {V : Type*} {P : Type*}
+
+/-- If two affine automorphisms agree on a set that spans the entire space, then they are equal.
+
+Specialization of `AffineMap.ext_on` to affine automorphisms. -/
+theorem ext_of_span_eq_top [CommRing k] [AddCommGroup V] [Module k V] [AffineSpace V P]
+    {s : Set P}
+    (h_span : affineSpan k s = ⊤)
+    (T₁ T₂ : P ≃ᵃ[k] P)
+    (h_agree : s.restrict T₁ = s.restrict T₂) : T₁ = T₂ := by
+  rw [← AffineEquiv.toAffineMap_inj]
+  apply AffineMap.ext_on h_span
+  exact fun x hx => congr_fun h_agree ⟨x, hx⟩
```

Comment:
Similar to the other lemma, I think we should:

- move this to `Mathlib.LinearAlgebra.AffineSpace.AffineSubspace.Basic`
- rename it to `AffineEquiv.ext_on`
- state it using `Set.eqOn`
- generalise it to an equivalence between two different spaces

It will then end up looking like:
```lean
theorem ext_on {s : Set P₁} (h_span : affineSpan k s = ⊤)
    (T₁ T₂ : P₁ ≃ᵃ[k] P₂) (h_agree : s.EqOn T₁ T₂) : T₁ = T₂ :=
  (AffineEquiv.toAffineMap_inj).mp <| AffineMap.ext_on h_span h_agree
```